doc/manual/manual.tex

```diff
@@ -4732,6 +4732,100 @@ \subsubsection{Averaging material properties}
+                                      ConstraintMatrix &current_constraints)
+  {
+    Quadrature<dim> quadrature (simulator_access.get_fe().get_unit_support_points());
+    FEValues<dim> fe_values (simulator_access.get_fe(), quadrature, update_q_points);
+    typename DoFHandler<dim>::active_cell_iterator cell;
+
+    for (cell = simulator_access.get_dof_handler().begin_active();
+           cell != simulator_access.get_dof_handler().end();
+           ++cell)
+      if (! cell->is_artificial())
+        {
+          fe_values.reinit (cell);
+          std::vector<unsigned int> local_dof_indices(simulator_access.get_fe().dofs_per_cell);
+          cell->get_dof_indices (local_dof_indices);
+
+          for (unsigned int q=0; q<quadrature.size(); q++)
+            // If it's okay to constrain this DOF
+            if (current_constraints.can_store_line(local_dof_indices[q]) &&
+                !current_constraints.is_constrained(local_dof_indices[q]))
+              {
+                // Get velocity component index
+                const unsigned int c_idx =
+                    simulator_access.get_fe().system_to_component_index(q).first;
+                // If we're on one of the velocity DOFs
+                if (c_idx >= simulator_access.introspection().component_indices.velocities[0] &&
+                    c_idx <= simulator_access.introspection().component_indices.velocities[dim-1])
+                  {
+                    // Which velocity component is this DOF associated with?
+                    const unsigned int component_direction = c_idx
+                        - simulator_access.introspection().component_indices.velocities[0];
+
+                    // Where are we in space?
+                    Point<dim> p = fe_values.quadrature_point(q);
+
+                    // Is this a place where we want to constrain the velocity?
+                    const bool indicator = (p[0] > .25 && p[0] < .75 && p[1] > .25 && p[1] < .75);
+                    // What velocity do we want to prescribe ?
+                    const double u_i = (component_direction == 0) ? .25 : -.25;
+
+                    if (indicator)
+                      {
+                        // Add line to mass matrix (if there isn't already one for this DOF)
+                        current_constraints.add_line (local_dof_indices[q]);
+                        // Add constraint to this DOF
+                        current_constraints.set_inhomogeneity (local_dof_indices[q], u_i);
+                      }
+                  }
+              }
+        }
+  }
+
+  // Connect constraints function to correct signal.
+  template <int dim>
+  void signal_connector (SimulatorSignals<dim> &signals)
+  {
+    signals.post_constraints_creation.connect (&constrain_internal_velocities<dim>);
+  }
+
+  // Tell Aspect to send signals to the connector functions
+  ASPECT_REGISTER_SIGNALS_CONNECTOR(signal_connector<2>, signal_connector<3>)
+}
+\end{lstlisting}
+
+
```

Comment:
do you still intend to write a section that shows and discussed the results (and includes the picture that's already part of the PR)?